backend/main.go

```diff
@@ -21,9 +22,18 @@ import (
 const timeFormat = "2006-01-02T15:04:05Z07:00"
 
 // sendResult sends a JSON payload to the given response writer
-func sendResult(w http.ResponseWriter, v interface{}) {
+func sendResult(w http.ResponseWriter, v interface{}, success bool) {
+	var statCode = 404
+	if success {
+		statCode = 200
+	}
 	// enable CORS
 	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Credentials", "true")
+	w.Header().Set("Access-Control-Allow-Methods", "GET,HEAD,OPTIONS,POST,PUT")
```

Comment:
We only really use POST calls since we want to send a request body, so I'm not 100% sure if we need the other methods.